web3_proxy/src/frontend/users/payment.rs

```diff
@@ -384,10 +385,17 @@ pub async fn user_balance_post(
             recipient_account, token, amount
         );
 
+        // Create a new transaction that will be used for joint transaction
+        let txn = db_conn.begin().await?;
+
+        // We must (1) lock the user and (2) lock the balance
+        // Both balance and lock must be present
+        // no need to do OnConflict with this functionality
         // Encoding is inefficient, revisit later
         let recipient = match user::Entity::find()
             .filter(user::Column::Address.eq(&recipient_account.encode()[12..]))
```

Comment:
Which "encode" function is this? Ethers types usually have "encode_hex" which includes the 0x prefix. If we are inconsistent about including the prefix, things will be broken.

Reply:
the database stores a Vec<u8>, and this is how I try to get to this, let me know if you still think it's wrong. this was working during a test that I ran, but can change it if you prefer another method

Reply:
we want to go from Address to Vec<u8> (specifically, [u8; 20], but as a Vec form because types must match). from_slice will not work here. I agree that there should be a cleaner way though